compiler/rustc_session/src/options.rs

```diff
@@ -85,6 +85,53 @@ pub struct TargetModifier {
 
 mod target_modifier_consistency_check {
     use super::*;
+    pub(super) fn target_feature(
+        sess: &Session,
+        l: &TargetModifier,
+        r: Option<&TargetModifier>,
+    ) -> bool {
+        fn parse<'a>(session: &'a Session, val: &'a TargetModifier) -> Vec<&'a str> {
+            let mut target_features = Vec::new();
+
+            for feature in session
+                .target_features
+                .iter()
+                .map(|symbol| symbol.as_str())
+                .chain(val.value_name.split(','))
+            {
+                if let Some(feature) = feature.strip_prefix('+') {
+                    if session
+                        .target
+                        .rust_target_features()
+                        .iter()
+                        .any(|(name, _, _, target_modifier)| feature == *name && *target_modifier)
+                    {
+                        target_features.push(feature);
+                    }
+                } else if let Some(feature) = feature.strip_prefix('-') {
+                    if session
+                        .target
+                        .rust_target_features()
+                        .iter()
+                        .any(|(name, _, _, target_modifier)| feature == *name && *target_modifier)
+                    {
+                        for (index, target_feature) in target_features.iter().enumerate() {
+                            if *target_feature == feature {
+                                target_features.remove(index);
+                                break;
+                            }
+                        }
+                    }
+                }
+            }
```

Comment:
I suspect this parsing already exists elsewhere in the codebase -- would it be possible to deduplicate the parsing routine?